File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
a, b, c = map(int, input().split())
while a != 0 and b != 0 and c != 0:
    data = [a, b, c]
    data.sort()
    if data[0] + data[1] <= data[2]:
        print("Invalid")
    elif data[0] == data[1] == data[2]:
        print("Equilateral")
    elif data[0] == data[1] or data[1] == data[2] or data[0] == data[2]:
        print("Isosceles")
    else:
        print("Scalene")

    a, b, c = map(int, input().split())    

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            int x = sc.nextInt();
            int y = sc.nextInt();
            int z = sc.nextInt();
            if (x == 0 && y == 0 && z == 0)
                break;

            System.out.println(checkLength(x, y, z));
        }

    }

    public static String checkLength(int x, int y, int z){
        int[] triangle = {x, y, z};
        Arrays.sort(triangle);
        if (x==y && y==z)
            return "Equilateral";
        else if (triangle[2] >= triangle[0] + triangle[1])
            return "Invalid";
        else if (x == y || y == z || x == z)
            return "Isosceles";
        else
            return "Scalene";
    }
}
